src/core/InstrumentPlayHandle.cpp

```diff
@@ -47,22 +47,20 @@ void InstrumentPlayHandle::play(sampleFrame * working_buffer)
 	// ensure that all our nph's have been processed first
 	auto nphv = NotePlayHandle::nphsOfInstrumentTrack(instrumentTrack, true);
 
-	bool nphsLeft = false;
-	do
+	for (const auto constNotePlayHandle : nphv)
 	{
-		for (const NotePlayHandle * constNotePlayHandle : nphv)
+		const auto notFinished = [&] {
+			return constNotePlayHandle->state() != ThreadableJob::ProcessingState::Done
+				&& !constNotePlayHandle->isFinished();
+		};
+
+		if (notFinished())
 		{
-			if (constNotePlayHandle->state() != ThreadableJob::ProcessingState::Done &&
-				!constNotePlayHandle->isFinished())
-			{
-				nphsLeft = true;
-				NotePlayHandle * notePlayHandle = const_cast<NotePlayHandle *>(constNotePlayHandle);
-				notePlayHandle->process();
-			}
+			const_cast<NotePlayHandle*>(constNotePlayHandle)->process();
+			while (notFinished()) {}
```

Comment:
I feel like this fundamentally changes the way it's processing from before.

Before:
Loop through all handles, if one is not done call it to process in its thread. Loop until done.

After:
Loop through all handles, if one is not done call it to process and spin wait until that single thread is done. Continue to the next handle.

Reply:
Oh you said in the description that's on purpose. 😁

Reply:
The previous behavior where it looped over all the handles multiple times didn't sit right with me. It felt inefficient, assuming we have a bit of handles in there.

Reply:
Before all the threads can awake and process. Now they are handled one by one only.

Reply:
Oh, I see now. I'll change it back.